chia/consensus/get_block_generator.py

```diff
@@ -0,0 +1,72 @@
+from __future__ import annotations
+
+from typing import Awaitable, Callable, Dict, Optional, Set
+
+from chia.types.block_protocol import BlockInfo
+from chia.types.blockchain_format.sized_bytes import bytes32
+from chia.types.full_block import FullBlock
+from chia.types.generator_types import BlockGenerator
+from chia.util.errors import Err
+from chia.util.ints import uint32
+
+
+async def get_block_generator(
```

Comment:
i think this can be part of FullBllocksProtocol ?
that way you can avoid passing in lookup_block_generators from the object implementing FullBllocksProtocol and just call get_block_generator with a block and additional blocks

Reply:
it makes it a lot harder to unit test in that case, because you need to mock the whole `FullBlocksProtocol`, rather than just the one function needed by `get_block_generator()`

Reply:
also, the protocols aren't base classes, so you would also need to duplicate this function in any class implementing the protocol. There's only one class that does right now, but I have future commits implementing it in `AugmentedBlockchain`